### DISCUSSION TITLE
default value in menu item

### DISCUSSION THREAD

**Discussion by rozrabiak** [forntoh/LcdMenu#161] [Ideas]
Hi, it is possible to change library to read value from variable when entering to menu/submenu?
My example:

```
bool relay = false;

MAIN_MENU(
    ITEM_BASIC("Start service"),
    ITEM_BASIC("Connect to WiFi"),
    ITEM_TOGGLE("Backlight", toggleBacklight, relay),     //in here relay is value to display
    ITEM_BASIC("Blink SOS"),
    ITEM_BASIC("Blink random")
);

void setup()
{
relay = digitalRead(pinNumber);  //or read settings from eeprom to variables
}
```

**Comment by rozrabiak** [comment]
I don't know how to pass value to ITEM_TOGGLE when menu.show() was called.. :/
And special: how to pass value when entering SUB_MENU...
My above example is "perfect"

**Comment by forntoh** [comment]
This discussion might be of help to you. If not, then comment here and I can look for other resources or guide you.

https://github.com/forntoh/LcdMenu/discussions/67

**Reply by rozrabiak** [reply]
Thank's for Your time.

ok, let me understand.
1. Can I use setValue anywhere in the code? Or only when I enter to item in menu/submenu?
For example: when relay was turned off automatically (after time has passed) - can I immediatelly use setValue for item in menu?
Menu must be show?

2. How to understand this entry:
```
menu[1][2][2].setValue("ON");
```

It is correct(?):
[1] - main menu
[2] - subMenu
[2] - item in submenu

**Reply by forntoh** [reply]
Yes, you're partially correct

`menu[1]` is the first item in main menu.

`menu[1][2]` is the second item in submenu at position 1 of the main menu. (So here, the first item in the main menu must be a submenu)

That's how it works. You can get update any menu item without having to navigate to the said item first. If you want to see the changes immediately then you perform an action that will trigger the menu to update or just call `menu.update()` after you are done.
